src/firebase_functions/options.py

```diff
@@ -223,32 +197,141 @@ def asdict_with_global_options(self) -> dict:
                 merged_options[key] = global_options[key]
         # None values are automatically stripped out in ManifestEndpoint generation.
```

Comment:
About that... we should sync. In the Node SDK we're now defaulting to using nulls for all unspecified options if a new option, preserveExistingChanges, is set to false (default). We will draft an API proposal for you and send it out, but you can assume it will pass. No need to make that change in this PR, just want to give a heads up.